### ISSUE TITLE
Compare validator returns valid when comparing true to false

### ISSUE THREAD

**Issue by benmccallum** [jeffhandley/strickland#15]
This might be by design so apologies upfront. Scenario: Terms and conditions checkbox. Must be ticked.

This returns `isValid: true` when value passed is `false`.
```javascript
acceptsTerms: compare({ compare: true, message: 'You must agree to Ts & Cs' })
```

It becomes necessary to do this, which if by design is fair enough:
```javascript
acceptsTerms: every(
  [required(), compare(true)],
  { message: 'You must agree to Ts & Cs' }
)
```

FYI: Passing value of `null` correctly comes back as `isValid: false`.

Cheers. Now starting to play with building these validator functions and definitions it's clear how powerful and composable everything is. Love it.

**Comment by jeffhandley** [comment]
Ooh, this is an interesting scenario for sure.  I do like the pattern of `required` being the only validator that ensures a value is present so that all other validators can be used against optional fields, aborting validation if there's no value.  But this is also an important use case and it's unfortunate that you fell into an accidentally valid result.

For the checkbox scenario though, you might be able to get by with only a `required` validator.  There is a note about that scenario here: https://strickland.io/docs/Validators/required.html

Will the following work?

``` js
acceptTerms: required({ message: 'You must agree to the Ts and Cs' })
```

If so, please let me know how you think this could be more discoverable or handled more gracefully by `compare`.

**Comment by benmccallum** [comment]
Looks like it's because of this: https://github.com/jeffhandley/strickland/blob/b079cb86489b86f3a2aa1469350013ae51d0c6b3/strickland/src/compare.js#L23-L30

I guess if the value the user is trying to compare against is a Boolean it should handle it straight up.

**Comment by benmccallum** [comment]
Hah, you jumped in at the same time. Yea it's definitely handled by a simple `required`. 

I wonder if the following would make sense:
```
if (typeof compare !== Boolean && isFalsyButNotZero(value)) {
    // Empty values are always valid except with the required validator

} else if (value !== compare) {
    isValid = false;
}
```

Where we just check the type of what the dev intended to compare against is, and if it's a Boolean, well let's just do a straight up value+type comparison with !==.

**Comment by benmccallum** [comment]
Just created a PR for that change (+ some linting). Feel free to disagree though if there could be conflicts.

**Comment by jeffhandley** [comment]
Fixed by #16
